src/main/java/tempest/GUI/GUIManager.java

```diff
@@ -0,0 +1,126 @@
+package tempest.GUI;
+
+import tempest.State;
+import tempest.Supervisor;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.Stack;
+
+//TODO:
+// - Change AddModule and AddSession pages to be Modules and Session pages with options to delete, view & add
+// - Do addModule() and deleteModule() need to be in the manager?
+
+public class GUIManager {
+    private static JPanel cardPanel;
+    private static CardLayout cl;
+
+    private HomePage home;
+    private AddModulePage addModule;
+    private AddSessionPage addSession;
+
+    private String currentCard;
+    private final Stack<String> cards = new Stack<>();
+
+    private final State state;
+    private final Supervisor supervisor;
+
+    public GUIManager(State state, Supervisor supervisor){
+        this.state = state;
+        this.supervisor = supervisor;
+        run();
+    }
+
+    /**
+     * Sets up and runs the GUI
+     */
+    private void run(){
```

Comment:
Could this changed to `start`, so we maintain the same terminology of `onStart` and `onClose`? Otherwise, I feel like it opens the door to confusion down the line.

Reply:
Done